lib/commands/run/index.js

```diff
@@ -0,0 +1,152 @@
+                return cb(tokenErr);
+            }
+            this._getSkillRunFlow(skillId, profile, skillCodeRegion, cmd.waitForAttach, watch, cmd.debug, debugPort, token,
+                runRegion, (err, runFlow) => {
+                    if (err) {
+                        Messenger.getInstance().error(err);
+                        return cb(err);
+                    }
+                    Messenger.getInstance()
+                        .info('\n*****Once the session is successfully started, '
+                            + 'you can use `ask dialog` to make simulation requests to your local skill code*****\n');
+                    if (cmd.waitForAttach) {
+                        Messenger.getInstance()
+                            .info(`\n*****Debugging session will wait until inspector is attached at port - ${debugPort}*****\n`);
+                    }
+                    runFlow.execCommand();
+                });
+        });
+    }
+
+    _getAccessTokenForProfile(profile, debug, callback) {
+        const authorizationController = new AuthorizationController({
+            auth_client_type: 'LWA',
+            doDebug: debug
+        });
+        authorizationController.tokenRefreshAndRead(profile, (tokenErr, token) => {
+            if (tokenErr) {
+                return callback(tokenErr);
+            }
+            callback(null, token);
+        });
+    }
+
+    _getHostedSkillRuntime(smapiClient, skillId, callback) {
+        smapiClient.skill.alexaHosted.getAlexaHostedSkillMetadata(skillId, (err, response) => {
+            if (err) {
+                return callback(err);
+            }
+            if (response.statusCode >= 300) {
+                const error = jsonView.toString(response.body);
+                return callback(error);
+            }
+            try {
+                const { runtime } = response.body.alexaHosted;
+                if (!stringUtils.isNonBlankString(runtime)) {
+                    throw new CliError(`Unable to determine runtime of the hosted skill - ${skillId}`);
+                }
+                callback(null, helper.getNormalisedRuntime(runtime));
+            } catch (error) {
+                return callback(error);
+            }
+        });
+    }
+
+    _getSkillRunFlow(skillId, profile, skillCodeRegion, waitForAttach, watch, debug, debugPort, token, runRegion, callback) {
+        if (this._filterAlexaHostedSkill(profile)) {
+            const smapiClient = new SmapiClient({
+                profile,
+                doDebug: debug
+            });
+            this._getHostedSkillRuntime(smapiClient, skillId, (err, runtime) => {
+                if (err) {
+                    return callback(err);
+                }
+                try {
+                    callback(null, helper.getSkillFlowInstance(runtime, helper.getHostedSkillInvocationInfo(runtime),
+                        waitForAttach, debugPort, token, skillId, runRegion, watch));
```

Comment:
there is usually a best practice when you have to use both trycatch and callback : don't mix the two together or some wierd issue might happen.

For this case, you can 
```
let skillFlow, hostedSkillInvocationInfo
try {
  skillFlow = helper.xxx
  hostedSkillInvocationInof = helper.yyy
} catch (e) {
  // handle e
}
callback(null, skillFlow, hostedSkillInvocationInof, ...)
```

Reply:
If surronding callback in try catch, the callback's err will potentially be caught again, and trigger the callback two times

Reply:
Good catch!